encoders/src/main/scala/org/apache/spark/sql/execution/columnar/SharedExternalStoreUtils.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ *
+ */
+package org.apache.spark.sql.execution.columnar
+
+import java.sql.{Connection, PreparedStatement, SQLException, Types}
+import java.util.Properties
+
+import scala.collection.immutable.HashMap
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Random
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.execution.ConnectionPool
+import org.apache.spark.sql.row.SnappyStoreClientDialect
+import org.apache.spark.sql.sources.ConnectionProperties
+import org.apache.spark.sql.types.{Decimal, StructType}
+
+object SharedExternalStoreUtils {
+
+  private var useLocatorURL: Boolean = _
+
+  final val COLUMN_BATCH_SIZE = "COLUMN_BATCH_SIZE"
+  final val COLUMN_MAX_DELTA_ROWS = "COLUMN_MAX_DELTA_ROWS"
+
+  private def useLocatorUrl(hostList: ArrayBuffer[(String, String)]): Boolean =
+    hostList.isEmpty
+
+  def getTableSchema(schemaAsJson: String): StructType = StructType.fromString(schemaAsJson)
+
+  def getConnection(connectionProperties: ConnectionProperties,
+      hostList: ArrayBuffer[(String, String)]): Connection = {
+    useLocatorURL = useLocatorUrl(hostList)
+    createConnection(connectionProperties, hostList)
+  }
+
+  private def createConnection(connProperties: ConnectionProperties,
+      hostList: ArrayBuffer[(String, String)]): Connection = {
+    // TODO CHECK THIS SOCKETCREATOR DEPENDENCY RESOLUTION
+    // val localhost = SocketCreator.getLocalHost
+    var index = -1
+
+    val jdbcUrl = if (useLocatorURL) {
+      connProperties.url
+    } else {
+      // TODO NEEDS TO BE HANDLED=
+      // if (index < 0) index = hostList.indexWhere(_._1.contains(localhost.getHostAddress))
+      if (index < 0) index = Random.nextInt(hostList.size)
+      hostList(index)._2
+    }
+
+    // TODO : REMOVE THIS AND ADD DEPENDENCY TO ACCESS ClientAttribute.THRIFT_LOB_DIRECT_BUFFERS
```

Comment:
ClientSharedUtils should be available. If not, then add gemfire-shared dependency which is fine (even snappydata-store-shared is fine).